Dockerfile

```
@@ -1,89 +1,45 @@
-FROM nvidia/cuda:11.8.0-cudnn8-devel-ubuntu22.04 as mava-core
-# Flag to record agents
-ARG record
-# Ensure no installs try launch interactive screen
+FROM nvidia/cuda:11.8.0-cudnn8-runtime-ubuntu22.04
+
+# Ensure no installs try to launch interactive screen
 ARG DEBIAN_FRONTEND=noninteractive
-# Update packages
+
+# Update packages and install python3.9 and other dependencies
 RUN apt-get update -y && \
     apt-get install -y software-properties-common && \
     add-apt-repository -y ppa:deadsnakes/ppa && \
-    apt install -y python3.9 && \
-    apt install -y python3.9-dev && \
-    apt-get install -y python3-pip && \
-    apt-get install -y python3.9-venv
-
-RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.9 10
+    apt-get install -y python3.9 python3.9-dev python3-pip python3.9-venv && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3.9 10 && \
+    python -m venv mava && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
-# Check python v
-RUN python -V
-
-# Setup virtual env
-RUN python -m venv mava
+# Setup virtual env and path
 ENV VIRTUAL_ENV /mava
 ENV PATH /mava/bin:$PATH
-RUN pip install --upgrade pip setuptools wheel
+
 # Location of mava folder
 ARG folder=/home/app/mava
-## working directory
+
+# Set working directory
 WORKDIR ${folder}
-## Copy code from current path
-COPY . /home/app/mava
-# For box2d
-RUN apt-get install swig -y
-## Install core dependencies + reverb.
-RUN pip install -e .[reverb]
-## Optional install for screen recording.
-ENV DISPLAY=:0
-RUN if [ "$record" = "true" ]; then \
-    ./bash_scripts/install_record.sh; \
-    fi
-EXPOSE 6006
-##########################################################
 
-# Jax Images
-##########################################################
-# Core Mava image
-FROM mava-core as jax-core
-# Jax gpu config.
-ENV XLA_PYTHON_CLIENT_PREALLOCATE=false
-## Install core jax dependencies.
-# Install jax gpu
-RUN pip install -e .[jax]
-# TODO (Ruan): This version is pinned now to avoid a breaking change in jaxlib.
-# Unpin when the issue is resolved.
-# Please see https://github.com/deepmind/dm-haiku/issues/565 for details.
-RUN pip install --upgrade "jax[cuda]==0.3.24" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
-##########################################################
+# Copy all code needed to install dependencies
+COPY ./requirements ./requirements
+COPY setup.py .
+COPY README.md .
+COPY mava/version.py mava/version.py
 
-##########################################################
-# PZ image
-FROM jax-core AS pz
-RUN pip install -e .[pz]
-# PettingZoo Atari envs
-RUN apt-get update
-RUN apt-get install ffmpeg libsm6 libxext6  -y
-RUN apt-get install -y unrar-free
-RUN pip install autorom
-RUN AutoROM -v
-##########################################################
+RUN echo "Installing requirements..."
+RUN pip install --quiet --upgrade pip setuptools wheel &&  \
+    pip install -e .
```

Not really sure what's best here, maybe we should copy jumanji, but what about:
```suggestion
    pip install -r requirements/requirements.txt
```

---

I tried this, but then it complains that `mava` isn't a module. I think this works on zathura since jumanji is added in `requirements.txt` there.